proxy/reverse.go

```diff
@@ -92,6 +96,16 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	removeSingleHopHeaders(&proxyreq.Header)
 	maybeSetForwardedFor(proxyreq)
 
+	if seen := checkProxyLoop(proxyreq, p.proxyID); seen {
+		p.director.Lock()
+		p.director.bannedEndpoints = p.endpoints
+		p.director.Unlock()
+
+		// update endpoints by calling refresh method
+		p.director.refresh()
+		plog.Debugf("found proxy loop (%v)", seen)
```

Comment:
Should we just return before we get to this point? `refresh` call in line 105 removes the looping endpoints so it shouldn't matter though.